### ISSUE TITLE
Failure to find py.typed for implicit namespace packages

### ISSUE THREAD

**Issue by simwr872** [microsoft/pyright#1343]
**Describe the bug**
Implicit namespace packages without a top-level `py.typed` file does not have its module `py.typed` file found.

**To Reproduce**
See code and folder structure below.

**Expected behavior**
According to the second to last paragraph of [https://www.python.org/dev/peps/pep-0561/#packaging-type-information](https://www.python.org/dev/peps/pep-0561/#packaging-type-information) this is the preferred way to distributing type information in namespace packages.
> For namespace packages (see [PEP 420](https://www.python.org/dev/peps/pep-0420)), the py.typed file should be in the submodules of the namespace, to avoid conflicts and for clarity.

**Screenshots or Code**
Below is a minimal example, both `venv/.../site-packages/namespaced/package/__init__.py` and `venv/.../site-packages/namespaced/package/py.typed` are empty.
* Folder structure (`venv/` is a virtual environment generated with `python -m venv venv`, dots denote various folders before `site-packages/`)
  ```
  main.py
  pyrightconfig.json
  venv/
  └─ ...
     └─ site-packages/
        └─ namespaced/
           └─ package/
              ├─ __init__.py
              └─ py.typed
  ```
* main.py
  ```python
  import namespaced.package
  print(namespaced.package)
  ```
* pyrightconfig.json
  ```json
  {
      "venvPath": ".",
      "venv": "venv",
      "typeCheckingMode": "strict"
  }
  ```
Running pyright results in the following report:
```
  1:8 - error: Stub file not found for "namespaced.package" (reportMissingTypeStubs)
  2:7 - error: Type of "package" is unknown (reportUnknownMemberType)
  2:7 - error: Argument type is unknown
    Argument corresponds to parameter "values" in function "print" (reportUnknownArgumentType)
```
Moving `py.typed` to one folder above does not produce any errors.

**VS Code extension or command-line**
Command-line using version `1.1.99`.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of pyright and pylance.

**Comment by erictraut** [comment]
This is fixed in Pyright 1.1.100, which I just published. It will also be included in the next release of Pylance.
